pkg/eestream/aesgcm.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package eestream
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+)
+
+type aesgcmEncrypter struct {
+	blockSize     int
+	key           [32]byte
+	startingNonce [12]byte
+	overhead      int
+	aesgcm        cipher.AEAD
+}
+
+// NewAesGcmEncrypter returns a Transformer that encrypts the data passing
+// through with key.
+//
+// startingNonce is treated as a big-endian encoded unsigned
+// integer, and as blocks pass through, their block number and the starting
+// nonce is added together to come up with that block's nonce. Encrypting
+// different data with the same key and the same nonce is a huge security
+// issue. It's safe to always encode new data with a random key and random
+// startingNonce. The monotonically-increasing nonce (that rolls over) is to
+// protect against data reordering.
+//
+// When in doubt, generate a new key from crypto/rand and a startingNonce
+// from crypto/rand as often as possible.
+func NewAesGcmEncrypter(key *[32]byte, startingNonce *[12]byte,
```

Comment:
this is a super nitpicky thing, but the go style guide prefers that acronyms are all caps, so this would be `NewAESGCMEncrypter` instead of `NewAesGcmEncrypter` to match the go style guide